stm32wl: Fix gpio irq handling

erwango opened the pull request:
EXTI LL API was buggy and was not returning correct source input for some GPIO lines.

- Fix this 
- Add an overlay to enable gpio_basic_api test on nucleo_wl55jc
- Fix openocd configuration

Fixes #42218

erwango committed:
tests/drivers: gpio_basic_api: Add overlay to test nucleo_wl55jc

Add and overlay to enable gpio api test on this board.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: nucleo_wl55jc: Fix openocd configuration

'connect_assert_srst' is preventing openocd to
connect to the board in debug mode.

To enable debug, fix openocd configuration to propose
two configurations:
- Debug compatible (default)
- Sleep mode compatible (commented)

Sleep mode compatible configuration should be used
when flashing over board's stock firmware.
A note has been added in board documentation to detail
this.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
west.yml: Use fixed STM32Cube STM32WL for exti

STM32Cube LL API for EXTI was buggy and returned
the wrong gpio port as IRQ source.

Point to the fixed version.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

FRASTM commented on dts/arm/st/wl/stm32wl.dtsi:
```diff
@@ -105,11 +105,11 @@
 		};
 
-		exti: interrupt-controller@58000800 {
+		exti: interrupt-controller@40010000 {
```
There are 2 peripherals : EXTI and SYSCFG.

In the /drivers/interrupt_controller/intc_exti_stm32.c,  the **EXTI peripheral**   is actually  registered @ 0x58000800 when used by the `stm32_exti_enable` or `stm32_exti_disable`  functions.
-->  LL_EXTI_EnableIT_0_31() etc  refers to the EXTI peripheral.


This differs for stm32wl (and others stm32 mcus like stm32wb) from the **SYSCFG peripheral**  which is registered @  0x40010000 and  configured  by the   gpio_stm32_set_exti_source function (in the  drivers/gpio/gpio_stm32.c)
-->  the LL_SYSCFG_SetEXTISource(port, line) refers to the SYSCFG peripheral.

erwango replied:
Right, I mixed both in the debug session. Will fix, txs

FRASTM reviewed (commented):
check of the exti periph register bank (comparing to stm32wb)

mnkp reviewed (approved):
LGTM. Though, @erwango if you have time you could update the commit message of "boards: nucleo_wl55jc: Fix openocd configuration" commit and explain shortly why the current configuration was wrong.

erwango commented:
> LGTM. Though, @erwango if you have time you could update the commit message of "boards: nucleo_wl55jc: Fix openocd configuration" commit and explain shortly why the current configuration was wrong.

This is right. I've reworked commit to provide more clarity and  also added a note in board documentation.
